onnxruntime/core/framework/transformer_memcpy.cc

```diff
@@ -43,12 +43,21 @@ bool TransformerMemcpyImpl::ModifyGraph(const KernelRegistryManager& kernel_regi
   // for initializers shared by different providers, create dups
   ProcessInitializers();
 
+  for (auto arg : graph_.GetInputs())
+    BuildDefsMapping(arg, kernel_registries);
```

Comment:
Could you provide more information for the issue you encountered? Add a test case for example. This suppose not the right place for the fix.

Reply:
What is the right place to copy inputs to CUDA memory?

Reply:
please refer to [https://github.com/Microsoft/onnxruntime/blob/439dbbada9546fa41d8abfe09d94b7129f934fd4/onnxruntime/core/session/onnxruntime_c_api.cc#L416](https://github.com/Microsoft/onnxruntime/blob/439dbbada9546fa41d8abfe09d94b7129f934fd4/onnxruntime/core/session/onnxruntime_c_api.cc#L416)

---
In reply to: [251656562](https://github.com/Microsoft/onnxruntime/pull/395#discussion_r251656562) [](ancestors = 251656562)

Reply:
I still confused how/why it happened

---
In reply to: [251946583](https://github.com/Microsoft/onnxruntime/pull/395#discussion_r251946583) [](ancestors = 251946583,251656562)

Reply:
As discussed this fix should be good for now.